controllers/front/callbackok.php

```diff
@@ -0,0 +1,117 @@
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+require_once(_PS_MODULE_DIR_.'/altapay/lib/altapay/altapay-php-sdk/lib/AltapayCallbackHandler.class.php');
+require_once(_PS_MODULE_DIR_.'/altapay/helpers.php');
+require_once(_PS_MODULE_DIR_.'/altapay/lib/altapay/altapay-php-sdk/lib/AltapayMerchantAPI.class.php');
+
+class AltapayCallbackokModuleFrontController extends ModuleFrontController
+{
+    /**
+     * Method to follow when callback ok is being triggered
+     * @throws AltapayMerchantAPIException
+     */
+    public function postProcess()
+    {
+        try {
+            $xml = Tools::getValue('xml');
+            $callbackHandler = new AltapayCallbackHandler();
+            $response = $callbackHandler->parseXmlResponse($xml);
+
+
+            $shopOrderId = $response->getPrimaryPayment()->getShopOrderId();
+
+            
+            //this lock prevents orders to be created twice.
+            $fp = fopen(_PS_MODULE_DIR_.'/altapay/controllers/front/lock.txt', 'r');
+            flock($fp, LOCK_EX);
+
+            // load the cart
+            $cart = getCartFromUniqueId($shopOrderId);
+            if (!Validate::isLoadedObject($cart)) {
+                $this->unlock($fp);
+                die('Could not load cart - exiting');
+            }
+
+            // load the customer
+            $customer = new Customer((int)$cart->id_customer);
+
+            // check if an order already exists
+            $order = getOrderFromUniqueId($shopOrderId);
+            if (Validate::isLoadedObject($order)) {
+                // an order has already been created from this cart - redirect
+                Tools::redirect('index.php?controller=order-confirmation&id_cart='.$cart->id.'&id_module='
+                .$this->module->id.'&id_order='.$this->module->currentOrder.'&key='.$customer->secure_key);
+            }
+
+            // handle success
+            if ($response->wasSuccessful()) {
+                $orderStatus=(int)Configuration::get('PS_OS_PAYMENT');
+                $paymentType=$response->getPrimaryPayment()->getAuthType();
+                $amountPaid=$response->getPrimaryPayment()->getCapturedAmount();
+                $captureStatus = Tools::getValue('require_capture');
+                $currencyPaid=Currency::getIdByIsoCode($response->getPrimaryPayment()->getCurrency());
+                $transactionID = Tools::getValue('transaction_id');
+                /*if payment type is 'payment' funds have not yet been captured,
+                so Altapay returns zero as the captured amount. Therefore we assume full payment has been authorized.*/
+                if ($paymentType=='payment') {
+                    $amountPaid = $cart->getOrderTotal(true, Cart::BOTH);
+                    $currencyPaid = new Currency($cart->id_currency);
+                } else if ($paymentType=='paymentAndCapture' && $captureStatus == 'true') {
```

Comment:
Consistent use of `elseif` over `else if` and spacing.
```suggestion
                } elseif ($paymentType === 'paymentAndCapture' && $captureStatus === 'true') {
```